pulp_rpm/app/models/repository.py

```diff
@@ -316,6 +324,21 @@ def finalize_new_version(self, new_version):
             except RepositoryVersion.DoesNotExist:
                 previous_version = None
 
+        # If requested, validate that the added rpms are signed appropriately
+        if self.allowed_pub_keys:
```

Comment:
This might be cleaner moved into a own method

Reply:
Done.